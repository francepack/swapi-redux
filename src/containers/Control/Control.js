import React from 'react'
import PropTypes from 'prop-types'

const Control = (props) => {
  // const { getPeople, getPlanets, getVehicles } = props
  return(
    <div className='btn-container'>
      <button >People</button>
      <button >Planets</button>
      <button >Vehicles</button>
    </div>
  )
}

// Control.propTypes = {
//   getPeople: PropTypes.func.isRequired,
//   getPlanets: PropTypes.func.isRequired,
//   getVehicles: PropTypes.func.isRequired
// }

export default Control