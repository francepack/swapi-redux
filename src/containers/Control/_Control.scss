.btn-container {
  display: flex;
  justify-content: space-around;
}

.btn-container button {
  background-color: $brownTan;
  border: 0;
  border-radius: 15px;
  color: $midnightGreen;
  font-weight: 700;
  height: 35px;
  letter-spacing: 1px;
  margin-top: 30px;
  width: 140px;

  &:focus {
    outline: none;
  }
  &:hover {
    background-color: $taupe;
    cursor: pointer;
  }
  &:active {
    background-color: $taupe;
  }
  &:focus {
    background-color: $taupe;
    color: $Gold;
  }
  //  buttons;
}