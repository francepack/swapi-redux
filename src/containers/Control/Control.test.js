import React from 'react';
import { enzyme, shallow } from 'enzyme';
import Control from './Control';

describe('Control', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Control />
    )
  });  
  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot()
  });
});