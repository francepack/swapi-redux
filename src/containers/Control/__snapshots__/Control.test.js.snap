// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contorl should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Control />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Link
          className="people-direct"
          to="/people"
        >
          <button>
            People
          </button>
        </Link>,
        <Link
          className="planets-direct"
          to="/planets"
        >
          <button>
            Planets
          </button>
        </Link>,
        <Link
          className="vehicles-direct"
          to="/vehicles"
        >
          <button>
            Vehicles
          </button>
        </Link>,
      ],
      "className": "btn-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <button>
            People
          </button>,
          "className": "people-direct",
          "to": "/people",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "People",
          },
          "ref": null,
          "rendered": "People",
          "type": "button",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <button>
            Planets
          </button>,
          "className": "planets-direct",
          "to": "/planets",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Planets",
          },
          "ref": null,
          "rendered": "Planets",
          "type": "button",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <button>
            Vehicles
          </button>,
          "className": "vehicles-direct",
          "to": "/vehicles",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Vehicles",
          },
          "ref": null,
          "rendered": "Vehicles",
          "type": "button",
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            className="people-direct"
            to="/people"
          >
            <button>
              People
            </button>
          </Link>,
          <Link
            className="planets-direct"
            to="/planets"
          >
            <button>
              Planets
            </button>
          </Link>,
          <Link
            className="vehicles-direct"
            to="/vehicles"
          >
            <button>
              Vehicles
            </button>
          </Link>,
        ],
        "className": "btn-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <button>
              People
            </button>,
            "className": "people-direct",
            "to": "/people",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "People",
            },
            "ref": null,
            "rendered": "People",
            "type": "button",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <button>
              Planets
            </button>,
            "className": "planets-direct",
            "to": "/planets",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Planets",
            },
            "ref": null,
            "rendered": "Planets",
            "type": "button",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <button>
              Vehicles
            </button>,
            "className": "vehicles-direct",
            "to": "/vehicles",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Vehicles",
            },
            "ref": null,
            "rendered": "Vehicles",
            "type": "button",
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Control should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Control />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Link
          className="people-direct"
          to="/people"
        >
          <button>
            People
          </button>
        </Link>,
        <Link
          className="planets-direct"
          to="/planets"
        >
          <button>
            Planets
          </button>
        </Link>,
        <Link
          className="vehicles-direct"
          to="/vehicles"
        >
          <button>
            Vehicles
          </button>
        </Link>,
      ],
      "className": "btn-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <button>
            People
          </button>,
          "className": "people-direct",
          "to": "/people",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "People",
          },
          "ref": null,
          "rendered": "People",
          "type": "button",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <button>
            Planets
          </button>,
          "className": "planets-direct",
          "to": "/planets",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Planets",
          },
          "ref": null,
          "rendered": "Planets",
          "type": "button",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <button>
            Vehicles
          </button>,
          "className": "vehicles-direct",
          "to": "/vehicles",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Vehicles",
          },
          "ref": null,
          "rendered": "Vehicles",
          "type": "button",
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            className="people-direct"
            to="/people"
          >
            <button>
              People
            </button>
          </Link>,
          <Link
            className="planets-direct"
            to="/planets"
          >
            <button>
              Planets
            </button>
          </Link>,
          <Link
            className="vehicles-direct"
            to="/vehicles"
          >
            <button>
              Vehicles
            </button>
          </Link>,
        ],
        "className": "btn-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <button>
              People
            </button>,
            "className": "people-direct",
            "to": "/people",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "People",
            },
            "ref": null,
            "rendered": "People",
            "type": "button",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <button>
              Planets
            </button>,
            "className": "planets-direct",
            "to": "/planets",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Planets",
            },
            "ref": null,
            "rendered": "Planets",
            "type": "button",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <button>
              Vehicles
            </button>,
            "className": "vehicles-direct",
            "to": "/vehicles",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Vehicles",
            },
            "ref": null,
            "rendered": "Vehicles",
            "type": "button",
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
